opentelemetry-instrumentation/src/opentelemetry/instrumentation/propagators.py

```diff
@@ -0,0 +1,121 @@
+
+This is part of an upcoming W3C spec and will eventually make it to the Otel spec.
+
+https://w3c.github.io/trace-context/#trace-context-http-response-headers-format
+"""
+
+import typing
+from abc import ABC, abstractmethod
+
+import opentelemetry.trace as trace
+from opentelemetry.context.context import Context
+from opentelemetry.propagators import textmap
+from opentelemetry.trace import format_span_id, format_trace_id
+
+_HTTP_HEADER_ACCESS_CONTROL_EXPOSE_HEADERS = "Access-Control-Expose-Headers"
+_BACK_PROPAGATOR = None
+
+
+def get_global_back_propagator():
+    return _BACK_PROPAGATOR
+
+
+def set_global_back_propagator(propagator):
+    global _BACK_PROPAGATOR  # pylint:disable=global-statement
+    _BACK_PROPAGATOR = propagator
+
+
+class DictHeaderSetter:
+    def set(self, carrier, key, value):  # pylint: disable=no-self-use
+        old_value = carrier.get(key, "")
+        if old_value:
+            value = "{0}, {1}".format(old_value, value)
+        carrier[key] = value
+
+
+default_setter = DictHeaderSetter()
+
+
+class FuncSetter:
+    def __init__(self, func):
+        self._func = func
+
+    def set(self, carrier, key, value):
+        self._func(carrier, key, value)
+
+
+class ResponsePropagator(ABC):
+    @abstractmethod
+    def _header_name(self) -> str:
+        pass
+
+    @abstractmethod
+    def _format_header(self, span_context) -> str:
+        pass
+
+    def inject(
+        self,
+        carrier: textmap.CarrierT,
+        context: typing.Optional[Context] = None,
+        setter: textmap.Setter = default_setter,
+    ) -> None:
+        """Injects SpanContext into the HTTP response carrier."""
+        span = trace.get_current_span(context)
+        span_context = span.get_span_context()
+        if span_context == trace.INVALID_SPAN_CONTEXT:
+            return
+
+        header_name = self._header_name()
+        setter.set(carrier, header_name, self._format_header(span_context))
+        setter.set(
+            carrier, _HTTP_HEADER_ACCESS_CONTROL_EXPOSE_HEADERS, header_name,
+        )
+
+
+class TraceResponsePropagator(ResponsePropagator):
+    """Experimental propagator that injects tracecontext into HTTP responses."""
+
+    def _header_name(self) -> str:  # pylint: disable=no-self-use
+        return "traceresponse"
+
+    def _format_header(
+        self, span_context
+    ) -> str:  # pylint: disable=no-self-use
+        return "00-{trace_id}-{span_id}-{:02x}".format(
+            span_context.trace_flags,
+            trace_id=format_trace_id(span_context.trace_id),
+            span_id=format_span_id(span_context.span_id),
+        )
+
+
+class ServerTimingResponsePropagator(ResponsePropagator):
+    def _header_name(self) -> str:  # pylint: disable=no-self-use
+        return "Server-Timing"
+
+    def _format_header(
+        self, span_context
+    ) -> str:  # pylint: disable=no-self-use
+        return 'traceparent;desc="00-{trace_id}-{span_id}-{:02x}"'.format(
```

Comment:
What's the different between the two propagators? Why is it called ServerTiming?

Reply:
I guess it comes from [this](https://www.w3.org/TR/server-timing/#the-server-timing-header-field)?

Reply:
Because it used the `Server-Timing` header. We don't strictly need Server-Timing propagator here TBH. We can remove it and let distros or users implement it themselves. I think it's pretty common to use this header for such purposes so I added it here.

Reply:
The header name makes sense, but wondering why the implementation is just the traceparent? Isn't the implementation supposed to be something to do with metrics? I'm also okay with removing this propagator.

Reply:
OK. I think it might be better to just remove it. Will do.